### PR TITLE
Gear 703 rounding hotfix

### PR DESCRIPTION
**Pull request by naterichman**
Fix rounding issue that caused embedded localizers to be split incorrectly

* Remove mean across archive before rounding.

### CODE REVIEW

File: utils/dicom/dicom_archive.py [hunk 1]
```diff
@@ -244,19 +269,11 @@ def contains_embedded_localizer(self):
...
-        # Applying a certain rounding threshold fixes this issue.
-        def apply_rounding(t):
-            return tuple(map(lambda x: round(float(x), TOLERANCE_ON_ImageOrientationPatient), t))
-        iop_tuple_list = list(map(apply_rounding, iop_tuple_list))
-
-        image_count = len(iop_tuple_list)
-        iop_tuple_set = set(iop_tuple_list)
-        nunique_iop = len(iop_tuple_set)
+        image_count = rounded_iops.shape[0]
+        nunique_iop = unique_iops.shape[0]
```
Comment (NPann) [inline utils/dicom/dicom_archive.py:276]: What a pain we have to maintain 2 codebase for this...

File: tests/unit_tests/test_dicom_archive.py [hunk 2]
```diff
@@ -35,4 +35,10 @@ def test_dicom_archive_class_validate():
...
             assert DicomArchive(not_dicom_path, extract_dir=temp_dir, validate=True)
-        assert 'failed to parse DICOMs' in str(err.value)
\ No newline at end of file
+        assert 'failed to parse DICOMs' in str(err.value)
+
+def test_contains_localizer_round():
+    im_arr = [[0.9989550114, 0.04570378363, 0, 0, 0, -1], [0.9989550114, 0.04570382088, 0, 0, 0, -1], [0.9989550114, 0.04570381716, 0, 0, 0, -1], [0.9989550114, 0.04570382088, 0, 0, 0, -1],[1, 0, 0, 0, 1, 0]]
+    out = [[-0.0, 0.009, 0.0, 0.0, -0.2, -0.2], [-0.0, 0.009, 0.0, 0.0, -0.2, -0.2], [-0.0, 0.009, 0.0, 0.0, -0.2, -0.2], [-0.0, 0.009, 0.0, 0.0, -0.2, -0.2], [0.001, -0.037, 0.0, 0.0, 0.8, 0.8]]
+
+    assert DicomArchive._round_iop(im_arr).tolist() ==  out
```
Comment (NPann) [inline tests/unit_tests/test_dicom_archive.py:44]: Maybe add a test for the `dicom_tag_value_dict` route as well?

**Review by NPann — APPROVED** [review]
LGTM. Just a few minor comments.

File: utils/dicom/dicom_archive.py (hunk 1 above)
Comment (naterichman) [reply utils/dicom/dicom_archive.py:276]: What do you mean?

File: tests/unit_tests/test_dicom_archive.py [hunk 3]
```diff
@@ -37,8 +39,32 @@ def test_dicom_archive_class_validate():
...
+        im_patch.header_dict.get.return_value = im
+        ims.append(im_patch)
+    archive.dataset_list = ims
+
+    rounded = archive.dicom_tag_value_dict('ImageOrientationPatient')
+    assert list(rounded.keys()) == [tuple(o) for o in np.unique(np.array(out),axis=0).tolist()]
+
+
+#def test_create_dicom_tag_value_dict_IOP():
+
```
Comment (NPann) [inline tests/unit_tests/test_dicom_archive.py:70]: remove comment?

File: tests/unit_tests/test_dicom_archive.py [hunk 4]
```diff
@@ -37,8 +39,32 @@ def test_dicom_archive_class_validate():
...
+    tag_vals.return_value = im_arr
+    ims = []
+    for im in im_arr:
+        im_patch = MagicMock()
+        im_patch.header_dict.get.return_value = im
+        ims.append(im_patch)
+    archive.dataset_list = ims
+
+    rounded = archive.dicom_tag_value_dict('ImageOrientationPatient')
+    assert list(rounded.keys()) == [tuple(o) for o in np.unique(np.array(out),axis=0).tolist()]
```
Comment (NPann) [inline tests/unit_tests/test_dicom_archive.py:66]: Thanks for adding this.

**Review by NPann — APPROVED** [review]

**Pull request merged**